packages/opentelemetry-metrics/src/export/types.ts

```diff
@@ -37,6 +37,15 @@ export interface Distribution {
   sum: number;
 }
 
+export interface Histogram {
+  buckets: {
```

Comment:
why is it plural if it is an object, could this be a `bucket` ?

Reply:
It is describing all of the buckets. For instance if you have three buckets (`<10`, `<20`, `<30`) with measurements of `[5, 5, 5, 5, 5, 15, 15, 15, 25]`, you would have:

```typescript
hist: Histogram = {
	buckets: {
		boundaries: [10, 20, 30],
		counts: [5, 3, 1],
	},
	sum: 95,
	count: 9
}
```

Reply:
@vmarchaud maybe you can add comments on the interface explaining that the boundaries are upper boundaries, the counts are the number of measurements in the bucket, and so on. I think this will not be clear to people who need to maintain this later if you are not here to walk them through it.

Reply:
hmm shouldn't we have then 

```typescript
hist: Histogram = {
	buckets: [
		{
			boundary: 10,
			count: 5,
		},
		{
			boundary: 20,
			count: 3,
		},
		{
			boundary: 30,
			count: 1,
		},
	],
	sum: 95,
	count: 9
}
```

Reply:
that would be one different way to do it, but I don't think one way is any more correct than another

Reply:
I agree that its current form feel non-javascript, it's because i've just ported the same logic from Go to JS. I would try to make it clearer

Reply:
I've added the following comment to explain why this structure is needed:
```
/**
   * Buckets are implemented using two different array:
   *  - boundaries contains every boundary (which are upper boundary for each slice)
   *  - counts contains count of event for each slice
   * 
   * Note that we'll always have n+1 (where n is the number of boundaries) slice
   * because we need to count event that are above the highest boundary. This is the
   * reason why it's not implement using array of object, because the last slice
   * dont have any boundary.
   * 
   * Example if we measure the values: [5, 30, 5, 40, 5, 15, 15, 15, 25]
   *  with the boundaries [ 10, 20, 30 ], we will have the following state:
   * 
   * buckets: {
   *	boundaries: [10, 20, 30],
   *	counts: [3, 3, 2, 1],
   * }
   */
```